executor.go

```diff
@@ -376,6 +397,105 @@ func containerIsPresent(containers []docker.APIContainers, containerId string) b
 	return false
 }
 
+// maybePullContainer checks if we need to pull a container and does so if needed.
+func (exec *sidecarExecutor) maybePullContainer(taskInfo *mesos.TaskInfo) error {
+	var shouldPullContainer bool
+
+	if !container.CheckImage(exec.client, taskInfo) {
+		shouldPullContainer = true
+	}
+
+	if taskInfo.Container.Docker.ForcePullImage != nil && *taskInfo.Container.Docker.ForcePullImage {
+		shouldPullContainer = true
+	}
+
+	log.Infof("Using image '%s'", taskInfo.Container.Docker.Image)
+
+	// Pull the image if it's stale/missing or we're told to force it
+	if shouldPullContainer {
+		err := container.PullImage(exec.client, taskInfo, exec.dockerAuth)
+		if err != nil {
+			return err
```

Comment:
`return fmt.Errorf("failed to pull image: %s", err)`

Reply:
Already comes with prefix that makes sense.